enterprise/backend/src/metabase_enterprise/serialization/api.clj

```diff
@@ -123,28 +124,32 @@
 (api/defendpoint POST "/export"
   "Serialize and retrieve Metabase instance.
 
-  Parameters:
-  - `dirname`: str, name of directory and archive file (default: `<instance-name>-<YYYY-MM-dd_HH-mm>`)
-  - `all_collections`: bool, serialize all collections (default: true, unless you specify `collection`)
-  - `collection`: array of int, db id of a collection to serialize
-  - `settings`: bool, if Metabase settings should be serialized (default: `true`)
-  - `data_model`: bool, if Metabase data model should be serialized (default: `true`)
-  - `field_values`: bool, if cached field values should be serialized (default: `false`)
-  - `database_secrets`: bool, if details how to connect to each db should be serialized (default: `false`)
-
-  Outputs .tar.gz file with serialization results and an `export.log` file.
-  On error just returns serialization logs."
+  Outputs `.tar.gz` file with serialization results and an `export.log` file.
+  On error outputs serialization logs directly."
   [:as {{:strs [all_collections collection settings data_model field_values database_secrets dirname]
          :or   {all_collections true
                 settings        true
                 data_model      true}}
         :query-params}]
-  {collection       [:maybe (ms/QueryVectorOf ms/PositiveInt)]
-   all_collections  [:maybe ms/BooleanValue]
-   settings         [:maybe ms/BooleanValue]
-   data_model       [:maybe ms/BooleanValue]
-   field_values     [:maybe ms/BooleanValue]
-   database_secrets [:maybe ms/BooleanValue]}
+  {dirname          (mu/with [:maybe string?]
```

Comment:
Is there another benefit of using mu/with, compared to writing the schema inline here like so?:

```
[:maybe {:description "name of directory and archive file (default: `<instance-name>-<YYYY-MM-dd_HH-mm>`)"}
  string?]
```

I find that version to be more concise and simpler (since it's just a value). I can see where mu/with would be cool for e.g. (mu/with ms/BooleanValue {:default false}), since you can't easily add to the properties there ^. Not sure about for the ones wrapped with `:maybe`, tho I may be missing something.

Reply:
Well main benefit to me is twofold: 1) the uniformity of the calls, so it's easy to scan with a cursory glance to see if anything is out of order; 2) the fact that you can easily change/copy-paste schemas without metadata (defaults/descriptions) being in a way.

So I'm looking at it more as a pattern than a necessity, but it feels nice. I agree that just putting a map in is more obvious, but still. WDYT?

Reply:
my 2 cents is that I like it raw. 

> Well main benefit to me is twofold: 1) the uniformity of the calls, so it's easy to scan with a cursory glance to see if anything is out of order;

maybe it's because the option is the 2nd arg? would it be nice if it's the last?

> the fact that you can easily change/copy-paste schemas without metadata (defaults/descriptions) being in a way.

I imagine you'll need to change the description anyway, so you probably end up copy-ing the whole `(mu/with)` block.

Reply:
I mean in the case when you want to change the schema, with `mu/with` you can edit the schema, without it - you'll have to extract your metadata, then put it back in in the correct spot and so on. What if your schema is just `string?`? What if it's `ms/BooleanValue`? You're out of luck basically. I thought about that `with` guy for the longest time and pretty sure it makes everything nicer. It's like `with-meta`, you know, metadata should live outside the entity.